workflow/controller/controller.go

```diff
@@ -437,7 +439,9 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 			}
 		case deletePod:
 			propagation := metav1.DeletePropagationBackground
-			err := pods.Delete(ctx, podName, metav1.DeleteOptions{PropagationPolicy: &propagation})
+			err := pods.Delete(ctx, podName, metav1.DeleteOptions{
+				PropagationPolicy:  &propagation,
+				GracePeriodSeconds: pointer.Int64Ptr(wfc.Config.PodGCGracePeriodSeconds)})
```

Comment:
this is 30s by default, so presumably, you'll make this longer?

Reply:
Yes, it's necessary to make this longer for certain scenarios.

Reply:
@terrytangyuan @alexec I am still trying to understand how this change waits before deleing pods, this call here deletes the pod setting the grace-period-seconds, which as far as I understand what will happen:

1. SIGTERM is sent to container
2. SIGKILL is sent if container does not gracefully shutdown within the grace-period

If my container exits immediately after the SIGTERM or in this case is not even running as it is completed how is the grace period helping to delay it's deletion?

Reply:
The above was resolved and implemented in #6168